src/ext/db/pool.cr

```diff
@@ -0,0 +1,28 @@
+module DB
+  class Pool(T)
+    def total : Array(T)
+      @total
+    end
+
+    private def build_resource : T
+      resource = @factory.call
+      @total << resource
+      @idle << resource
+      if resource.is_a?(DB::Connection)
+        ConnectionStartedEvent.publish(connection: resource)
+      end
```

Comment:
Just added the pubsub to this so that we could go ahead and start/track a transaaction on any new connections.